### PR TITLE
Implement 5 BLACK_TEMPLE cards

### PR DESCRIPTION
**Pull request by scottsuk0306**
This revision includes
- Implementation of 5 BLACK_TEMPLE cards
  - 'Cursed Vagrant' (BT_703)
  - 'Enhanced Dreadlord' (BT_304)
  - 'Fel Summoner' (BT_509)
  - 'Coilfang Warlord' (BT_761)
  - 'Rustsworn Cutlist' (BT_160)

### CODE REVIEW

File: Sources/Rosetta/PlayMode/CardSets/BlackTempleCardsGen.cpp [hunk 1]
```diff
@@ -1803,6 +1819,16 @@ void BlackTempleCardsGen::AddDemonHunter(std::map<std::string, CardDef>& cards)
...
     //  - DEATHRATTLE = 1
     // --------------------------------------------------------
+    power.ClearData();
+    power.AddDeathrattleTask(std::make_shared<IncludeTask>(EntityType::HAND));
+    power.AddDeathrattleTask(std::make_shared<FilterStackTask>(SelfCondList{ std::make_shared<SelfCondition>(
+            SelfCondition::IsRace(Race::DEMON)) }));
+    power.AddDeathrattleTask(
+        std::make_shared<RandomTask>(EntityType::STACK, 1));
+    power.AddDeathrattleTask(std::make_shared<GetGameTagTask>(
+        EntityType::STACK, GameTag::ENTITY_ID));
```
Comment (utilForever) [inline Sources/Rosetta/PlayMode/CardSets/BlackTempleCardsGen.cpp:1829]: Is there a reason why you wrote this code?

File: Sources/Rosetta/PlayMode/CardSets/BlackTempleCardsGen.cpp [hunk 2]
```diff
@@ -2007,6 +2040,15 @@ void BlackTempleCardsGen::AddNeutral(std::map<std::string, CardDef>& cards)
     // RefTag:
     //  - DEATHRATTLE = 1
     // --------------------------------------------------------
+    power.ClearData();
+    power.AddPowerTask(
+        std::make_shared<IncludeTask>(EntityType::MINIONS_NOSOURCE));
+    power.AddPowerTask(std::make_shared<FilterStackTask>(
+        SelfCondList{ std::make_shared<SelfCondition>(
+            SelfCondition::IsMinion()) }));
```
Comment (utilForever) [inline Sources/Rosetta/PlayMode/CardSets/BlackTempleCardsGen.cpp:2048]: Is there a reason why you wrote this code?

File: Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp [hunk 3]
```diff
@@ -132,6 +132,121 @@ TEST_CASE("[Priest : Minion] - BT_258 : Imprisoned Homunculus")
...
+    game.Process(curPlayer, PlayCardTask::Minion(card1));
+    CHECK_EQ(curField.GetCount(), 1);
+    CHECK_EQ(curField[0]->card->name, "Cursed Vagrant");
+
+    game.Process(curPlayer, EndTurnTask());
+    game.ProcessUntil(Step::MAIN_ACTION);
+
+    game.Process(opPlayer, PlayCardTask::SpellTarget(card2, card1));
+    CHECK_EQ(curField[0]->card->name, "Cursed Shadow");
+    CHECK_EQ(curField.GetCount(), 1);
```
Comment (utilForever) [inline Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp:183]: You'd better change the position of these two lines.

File: Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp [hunk 4]
```diff
@@ -132,6 +132,121 @@ TEST_CASE("[Priest : Minion] - BT_258 : Imprisoned Homunculus")
...
+    CHECK_EQ(curField[0]->card->name, "Enhanced Dreadlord");
+    CHECK_EQ(curField[0]->HasTaunt(), true);
+
+    game.Process(curPlayer, EndTurnTask());
+    game.ProcessUntil(Step::MAIN_ACTION);
+
+    game.Process(opPlayer, PlayCardTask::SpellTarget(card2, card1));
+    game.Process(opPlayer, PlayCardTask::SpellTarget(card3, card1));
+    CHECK_EQ(curField[0]->card->name, "Desperate Dreadlord");
+    CHECK_EQ(curField.GetCount(), 1);
```
Comment (utilForever) [inline Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp:244]: You'd better change the position of these two lines.

File: Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp [hunk 5]
```diff
@@ -374,6 +489,118 @@ TEST_CASE("[Warrior : Spell] - BT_124 : Corsair Cache")
...
+    game.Process(curPlayer, PlayCardTask::Minion(card1));
+    CHECK_EQ(curField.GetCount(), 1);
+    CHECK_EQ(curField[0]->card->name, "Fel Summoner");
+
+    game.Process(curPlayer, EndTurnTask());
+    game.ProcessUntil(Step::MAIN_ACTION);
+
+    game.Process(opPlayer, PlayCardTask::SpellTarget(card5, card1));
+    CHECK_EQ(curField.GetCount(), 1);
+    CHECK_EQ(curField[0]->IsRace((Race::DEMON)), true);
```
Comment (utilForever) [inline Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp:543]:
```suggestion
    CHECK_EQ(curField[0]->IsRace(Race::DEMON), true);
```

File: Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp [hunk 6]
```diff
@@ -374,6 +489,118 @@ TEST_CASE("[Warrior : Spell] - BT_124 : Corsair Cache")
...
+    CHECK_EQ(curField.GetCount(), 1);
+    CHECK_EQ(curField[0]->card->name, "Coilfang Warlord");
+    CHECK_EQ(curField[0]->HasRush(), true);
+
+    game.Process(curPlayer, EndTurnTask());
+    game.ProcessUntil(Step::MAIN_ACTION);
+
+    game.Process(opPlayer, PlayCardTask::SpellTarget(card2, card1));
+    CHECK_EQ(curField[0]->card->name, "Conchguard Warlord");
+    CHECK_EQ(curField.GetCount(), 1);
```
Comment (utilForever) [inline Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp:598]: You'd better change the position of these two lines.

File: Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp [hunk 7]
```diff
@@ -681,4 +908,62 @@ TEST_CASE("[Neutral : Minion] - BT_159 : Terrorguard Escapee")
...
+    auto& curField = *(curPlayer->GetFieldZone());
+
+    const auto card1 = Generic::DrawCard(
+        curPlayer, Cards::FindCardByName("Rustsworn Cultist"));
+    const auto card2 = Generic::DrawCard(
+        curPlayer, Cards::FindCardByName("Wolfrider"));
+    const auto card3 =
+        Generic::DrawCard(opPlayer, Cards::FindCardByName("Fireball"));
+    const auto card4 =
+        Generic::DrawCard(opPlayer, Cards::FindCardByName("Fireball"));
```
Comment (utilForever) [inline Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp:956]: I think that this code is unused.

File: Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp [hunk 8]
```diff
@@ -681,4 +908,62 @@ TEST_CASE("[Neutral : Minion] - BT_159 : Terrorguard Escapee")
...
+
+    game.Process(curPlayer, PlayCardTask::Minion(card2));
+    game.Process(curPlayer, PlayCardTask::Minion(card1));
+
+    game.Process(curPlayer, EndTurnTask());
+    game.ProcessUntil(Step::MAIN_ACTION);
+
+    game.Process(opPlayer, PlayCardTask::SpellTarget(card3, card2));
+    CHECK_EQ(curField[0]->card->name, "Rusted Devil");
+    CHECK_EQ(curField.GetCount(), 2);
```
Comment (utilForever) [inline Tests/UnitTests/PlayMode/CardSets/BlackTempleCardsGenTests.cpp:966]: You'd better change the position of these two lines.

**Review by utilForever — CHANGES_REQUESTED** [review]
LGTM. However, you need to change some codes. Please check them. 😀

File: Sources/Rosetta/PlayMode/CardSets/BlackTempleCardsGen.cpp (hunk 1 above)
Comment (scottsuk0306) [reply Sources/Rosetta/PlayMode/CardSets/BlackTempleCardsGen.cpp:1829]: I think I misplaced the code. I'll delete it.

File: Sources/Rosetta/PlayMode/CardSets/BlackTempleCardsGen.cpp (hunk 2 above)
Comment (scottsuk0306) [reply Sources/Rosetta/PlayMode/CardSets/BlackTempleCardsGen.cpp:2048]: I added this part to specify "other minions" but I just noticed the line before does that job. I'll delete it.

**Review by utilForever — APPROVED** [review]
LGTM now. Thanks for the work! 🍕

**Pull request merged**